protocol/images/server.go

```diff
@@ -0,0 +1,108 @@
+package images
+
+import (
+	"context"
+	"database/sql"
+	"net"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/protocol/identity/identicon"
+)
+
+type messageHandler struct {
+	db     *sql.DB
+	logger *zap.Logger
+}
+
+type identiconHandler struct {
+	logger *zap.Logger
+}
+
+func (s *identiconHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	pks, ok := r.URL.Query()["publicKey"]
+	if !ok || len(pks) == 0 {
+		s.logger.Error("no publicKey")
+		return
+	}
+	pk := pks[0]
+	image, err := identicon.Generate(pk)
+	if err != nil {
+		s.logger.Error("could not generate identicon")
+	}
+
+	w.Header().Set("Content-Type", "image/png")
+	_, err = w.Write(image)
+	if err != nil {
+		s.logger.Error("failed to write image", zap.Error(err))
+	}
+
+}
+
+func (s *messageHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	messageIDs, ok := r.URL.Query()["messageId"]
+	if !ok || len(messageIDs) == 0 {
+		s.logger.Error("no messageID")
+		return
+	}
+	messageID := messageIDs[0]
+	var image []byte
+	err := s.db.QueryRow(`SELECT image_payload FROM user_messages WHERE id = ?`, messageID).Scan(&image)
+	if err != nil {
+		s.logger.Error("failed to find image", zap.Error(err))
+		return
+	}
+	if len(image) == 0 {
+		s.logger.Error("empty image")
+		return
+	}
+	mime, err := ImageMime(image)
+	if err != nil {
+		s.logger.Error("failed to get mime", zap.Error(err))
+	}
+
+	w.Header().Set("Content-Type", mime)
+	_, err = w.Write(image)
+	if err != nil {
+		s.logger.Error("failed to write image", zap.Error(err))
+	}
+}
+
+type Server struct {
+	Port   int
+	server *http.Server
+	logger *zap.Logger
+	db     *sql.DB
+}
+
+func NewServer(db *sql.DB, logger *zap.Logger) *Server {
+	return &Server{db: db, logger: logger}
+
+}
+
+func (s *Server) Start() error {
+	listener, err := net.Listen("tcp", ":0")
```

Comment:
Would it also be possible to use net.http.ListenAndServeTLS instead or a [tls.listener](https://pkg.go.dev/crypto/tls#NewListener)? TLS with self signed certs would harden the architecture here, also it wouldn't be a problem if at some point in time the server is moved out of the local application.

Reply:
Thanks for the review, sorry for the late reply, I missed the comment.
I'll look into it, we can definitely use a self signed certificate in the go code, but the tricky bit is to have the client accept a self signed certificate, I'll have a look.
So far I was looking at https://developer.apple.com/documentation/foundation/url_loading_system/handling_an_authentication_challenge/performing_manual_server_trust_authentication for IOS, but I'll have to spend a bit more time researching.